### DISCUSSION TITLE
Android Support

### DISCUSSION THREAD

**Discussion by chrisoutwright** [coddingtonbear/obsidian-local-rest-api#64] [General]
Request for Android Support in the Plugin:  I would like to request the addition of Android support to the Obsidian Local REST API plugin.     

I believe that adding Android support to the Obsidian Local REST API plugin would greatly enhance the overall user experience, broaden the plugin's reach, and empower users to unlock the full potential of Obsidian on their Android devices. Chromium-based browsers offer support for the plugin, getting the rest to work locally would be a plus.

What are the current challenges?

**Comment by coddingtonbear** [comment]
The main limiting factor is that we're relying on some node APIs that aren't, as far as I'm aware, available on mobile devices -- particularly for opening a port and launching a socket server.  I don't regularly use Obsidian on an android device, so I can't be sure, but I'm at least sure that getting this library running on an IOS device is essentiallly impossible.

**Reply by chrisoutwright** [reply]
Thank you for the answer.
I see that the lifecycle management policies could interfere with the functioning of Node.js APIs, which are designed to work in a constantly-running server environment. So may not be enough to do follow an approach involving creating a bridge or a wrapper layer that translates the Node.js APIs into equivalent APIs that can run on Android.

Essentially it would translate to: bypass the necessity of setting up a local REST API server and still share data to Obsidian using Android's 'Share to' functionality. Therefore sending 'intents' to Obsidian with the data to share (on obsidian side: involve registering an 'intent filter' within Obsidian's Android manifest to declare that it can accept shared data.). But this would be a whole different plugin then.

 Local REST on android seems really not feasible for that purpose, right? Constrained by: Android apps don't have the capability to open a port and listen for data when the app is in the background .
